libs/librrgraph/src/base/check_rr_graph.cpp

```diff
@@ -1,22 +1,24 @@
 #include "vtr_log.h"
 #include "vtr_memory.h"
+#include "vtr_util.h"
 
-#include "vpr_types.h"
 #include "vpr_error.h"
-
-#include "globals.h"
-#include "rr_graph.h"
 #include "check_rr_graph.h"
 
+#include "rr_node.h"
+#include "physical_types_util.h"
+
+#include "describe_rr_node.h"
+
 /*********************** Subroutines local to this module *******************/
 
-static bool rr_node_is_global_clb_ipin(RRNodeId inode);
+static bool rr_node_is_global_clb_ipin(const RRGraphView& rr_graph, const DeviceGrid& grid, RRNodeId inode);
 
-static void check_unbuffered_edges(int from_node);
+static void check_unbuffered_edges(const RRGraphView& rr_graph, int from_node);
 
-static bool has_adjacent_channel(const t_rr_node& node, const DeviceGrid& grid);
+static bool has_adjacent_channel(const RRGraphView& rr_graph, const DeviceGrid& grid, const t_rr_node& node);
```

Comment:
Just a mark here. This function need a refactoring. We should use ``RRNodeId`` rather than ``t_rr_node`` which is now an internal data of RRGraph